main_windows.go

```
@@ -30,11 +36,52 @@ var (
 
 	history      []string
 	historyIndex = 0
+
+	bookmarksFile string
+	bookmarks     []bookmark
+	bookmarksMenu *walk.Menu
 )
 
+type bookmark struct {
+	Address  string       `json:"address"`
+	Password string       `json:"password"`
+	Action   *walk.Action `json:"-"`
+}
+
 func init() {
 	kernel32, guiInitErr = syscall.LoadDLL("kernel32.dll")
 	freeConsole, guiInitErr = kernel32.FindProc("FreeConsole")
+
+	// Creates an folder in user home called "IceCon"
+	initStorage()
+	loadBookmarks()
+}
+
+func initStorage() {
+	// If user found create in user home
+	if usr, err := user.Current(); err == nil {
+		directory := path.Join(usr.HomeDir, "IceCon")
+		bookmarksFile = path.Join(usr.HomeDir, "Icecon", "bookmarks.json")
+
+		os.Mkdir(directory, os.ModePerm)
+	}
+}
+
+func loadBookmarks() {
+	// Loads bookmarks.json from IceCon directory
+	file, err := ioutil.ReadFile(bookmarksFile)
+	if err != nil {
+		return
```

Errors should definitely not be silently discarded, the user should instead be notified about this condition (either through the log or via a message box).